staging/src/k8s.io/api/events/v1/types.go

```diff
@@ -39,29 +40,29 @@ type Event struct {
 
 	// reportingController is the name of the controller that emitted this Event, e.g. `kubernetes.io/kubelet`.
 	// This field cannot be empty for new Events.
-	// +optional
-	ReportingController string `json:"reportingController,omitempty" protobuf:"bytes,4,opt,name=reportingController"`
+	// required.
+	ReportingController string `json:"reportingController" protobuf:"bytes,4,opt,name=reportingController"`
 
 	// reportingInstance is the ID of the controller instance, e.g. `kubelet-xyzf`.
 	// This field cannot be empty for new Events and it can have at most 128 characters.
-	// +optional
-	ReportingInstance string `json:"reportingInstance,omitempty" protobuf:"bytes,5,opt,name=reportingInstance"`
+	// required.
+	ReportingInstance string `json:"reportingInstance" protobuf:"bytes,5,opt,name=reportingInstance"`
 
 	// action is what action was taken/failed regarding to the regarding object. It is machine-readable.
 	// This field can have at most 128 characters.
-	// +optional
-	Action string `json:"action,omitempty" protobuf:"bytes,6,name=action"`
+	// required.
+	Action string `json:"action" protobuf:"bytes,6,name=action"`
 
 	// reason is why the action was taken. It is human-readable.
 	// This field can have at most 128 characters.
-	// +optional
-	Reason string `json:"reason,omitempty" protobuf:"bytes,7,name=reason"`
+	// required.
+	Reason string `json:"reason" protobuf:"bytes,7,name=reason"`
 
 	// regarding contains the object this Event is about. In most cases it's an Object reporting controller
 	// implements, e.g. ReplicaSetController implements ReplicaSets and this event is emitted because
 	// it acts on some changes in a ReplicaSet object.
-	// +optional
-	Regarding corev1.ObjectReference `json:"regarding,omitempty" protobuf:"bytes,8,opt,name=regarding"`
+	// required.
```

Comment:
I don't see this being required by validation, revert the changes to this field?

The following event can be created successfully:
```yaml
kind: Event
apiVersion: events.k8s.io/v1
metadata:
  generateName: foo
  namespace: kube-system
eventTime: "2017-08-09T15:04:05.000000Z"
reportingInstance: foo
reportingController: foo
action: Foo
reason: Foo
type: Warning
```

Reply:
Yes - this is definitely optional.

Reply:
Got it :) 
Actually, this fields were mentioned in the issue as supposedly required ones so I seem to follow that. My bad.
Anyways I have pushed the new changes as per your suggestions. 😊